src/main/java/com/microsoft/azure/functions/worker/handler/FunctionEnvironmentReloadRequestHandler.java

```diff
@@ -39,40 +46,53 @@ FunctionMethodDescriptor createFunctionDescriptor(String functionId,
   }
   
   /*
-   * This is a helper utility specifically to reload environment variables 
-   * if java language worker is started in standby mode by the functions runtime
+   * This is a helper utility specifically to reload environment variables and
+   * ReinitializeDetours if java language worker is started in standby mode by the functions runtime
    * and should not be used for other purposes
    */
-  public static void setEnv(Map<String, String> newSettings) throws Exception {
+  public void specialize(Map<String, String> newSettings) throws Exception {
     if (newSettings == null || newSettings.isEmpty()) {    
       return;
     }
-	  try {
-	    Class<?> processEnvironmentClass = Class.forName("java.lang.ProcessEnvironment");
-	    Field theEnvironmentField = processEnvironmentClass.getDeclaredField("theEnvironment");
-	    theEnvironmentField.setAccessible(true);
-	    Map<String, String> env = (Map<String, String>) theEnvironmentField.get(null);
-	    env.putAll(newSettings);
-	    Field theCaseInsensitiveEnvironmentField = processEnvironmentClass.getDeclaredField("theCaseInsensitiveEnvironment");
-	    theCaseInsensitiveEnvironmentField.setAccessible(true);
-	    Map<String, String> cienv = (Map<String, String>)     theCaseInsensitiveEnvironmentField.get(null);
-	    cienv.putAll(newSettings);
-	  } catch (NoSuchFieldException e) {
-	    Class[] classes = Collections.class.getDeclaredClasses();
-	    Map<String, String> env = System.getenv();
-	    for(Class cl : classes) {
-	      if("java.util.Collections$UnmodifiableMap".equals(cl.getName())) {
-	        Field field = cl.getDeclaredField("m");
-	        field.setAccessible(true);
-	        Object obj = field.get(env);
-	        Map<String, String> map = (Map<String, String>) obj;
-	        map.clear();
-	        map.putAll(newSettings);
-	      }
-	    }
-	  }
-	}
 
+    // This is required to reinitialize file detours
+    newSettings.put(WebsitePlaceholderMode, "0");
```

Comment:
Please note we will have to call Kernel32.INSTANCE.SetEnvironmentVariable(entry.getKey(), entry.getValue()); for all these environment variables so the process env variable is updated not only JVM env variable.
What you updated here only updates what's visible to JVM. That's why I put the code there to call  Kernel32.INSTANCE.SetEnvironmentVariable

Reply:
> for all these environment variables so the process env variable is updated not only JVM env variable.
Can you please clarify process vs JVM? are you referring to system environment variables? they would be set by the parent - functions host process right? I am trying to establish a pattern for specialization for other language workers as well.

Reply:
We used ProcessExplorer and noticed with the existing code, not all env variables were updated. so that's why added the new lines. If you can verify using ProcessExplorer that whatever is sent is added/updated and not existing ones get deleted then we should be fine

Reply:
Thanks! Existing code was intentionally overriding system environment variables with the new dictionary sent by the functions host. Will update the code to only add new keys and update existing keys

Reply:
And please don't forget to delete the ones which no longer exist.

Reply:
Existing code is already doing this. Deletes existing system environment variables and replaces them with the new dictionary sent by the functions host.

> We used ProcessExplorer and noticed with the existing code, not all env variables were updated. 

It is possible that environment variables sent by the functions host are updated only after specialization completes. For example `WEBSITE_PLACEHOLDER_MODE`  is updated by the host after specialization completes but the call to specialize language worker happens before. We should explicitly set these in the language worker.

Reply:
Ok I remember the site name was also still the mawsplaceholder.... instead of actual site name when looking at env variables. but when I added new code it got fixed to the actual site name. we can test that as well.